### PR TITLE
improvement: remove unused imports on Scala 3.4+

### PR DESCRIPTION
**Pull request by bjaglin**
Follows https://github.com/scalacenter/scalafix/pull/1800 / https://github.com/scalameta/metals/pull/6199

Users on Scala 3.4+ with  `-Wunused:import` in their build will get unused imports cleaned up

### CODE REVIEW

File: metals/src/main/scala/scala/meta/internal/metals/ScalafixProvider.scala [hunk 1]
```diff
@@ -269,6 +269,21 @@ case class ScalafixProvider(
       .getOrElse("Unexpected error while running Scalafix.")
   }
 
+  private lazy val scala3RemoveUnusedDefaultConfig = {
+    val path = Files.createTempFile(".scalafix", ".conf")
+    AbsolutePath(path).writeText(
+      s"""|rules = [
+          |  OrganizeImports
+          |]
+          |OrganizeImports.removeUnused = true
```
Comment (bjaglin) [inline metals/src/main/scala/scala/meta/internal/metals/ScalafixProvider.scala:278]: that's the default, I just made it explicit

File: metals/src/main/scala/scala/meta/internal/metals/ScalafixProvider.scala [hunk 2]
```diff
@@ -397,8 +425,11 @@ case class ScalafixProvider(
       val scalacOptions = {
         val list = new ju.ArrayList[String](3)
 
-        if (scalaBinaryVersion == "2.13") list.add("-Wunused:imports")
-        else list.add("-Ywarn-unused-import")
+        scalaBinaryVersion match {
+          case "3" => list.add("-Wunused:import")
+          case "2.13" => list.add("-Wunused:imports")
+          case _ => list.add("-Ywarn-unused-import")
```
Comment (bjaglin) [inline metals/src/main/scala/scala/meta/internal/metals/ScalafixProvider.scala:431]: From what I see, these are just passed to make `OrganizeImports` happy, but we are not sure they are actually sent to the compiler. Unrelated to this PR, but I wonder how/if we could let the user know when this is not the case?

File: metals/src/main/scala/scala/meta/internal/metals/ScalafixProvider.scala [hunk 3]
```diff
@@ -344,6 +366,13 @@ case class ScalafixProvider(
       rules: List[String],
   ): Future[ScalafixEvaluation] = {
     val isScala3 = ScalaVersions.isScala3Version(scalaTarget.scalaVersion)
+    val isSource3 = scalaTarget.scalac.getOptions().contains("-Xsource:3")
+    val isScala3Dialect = isScala3 || isSource3
+    val canRemoveUnused = !isScala3 ||
+      // https://github.com/scala/scala3/pull/17835
+      Seq("3.0", "3.1", "3.2", "3.3")
```
Comment (bjaglin) [inline metals/src/main/scala/scala/meta/internal/metals/ScalafixProvider.scala:373]: I took note on the scalafix PR to update that whenever this is backported to 3.3 LTS. We can't do it upfront as Scalafix has a similar check.

File: metals/src/main/scala/scala/meta/internal/metals/ScalafixProvider.scala (hunk 2 above)
Comment (tgodzik) [reply metals/src/main/scala/scala/meta/internal/metals/ScalafixProvider.scala:431]: > Unrelated to this PR, but I wonder how/if we could let the user know when this is not the case?

We don't really report that, might be good to do that as a follow up.

**Review by tgodzik — APPROVED** [review]
Thanks very much for all your work with scalafix!

**Pull request merged**